File: trunk/PolarMath/src/org/macondo/polar/evaluation/hrv/AMoPercents.java
package org.macondo.polar.evaluation.hrv;

import java.util.List;

import org.macondo.polar.data.Training;
import org.macondo.polar.evaluation.Evaluation;
import org.macondo.polar.util.Histogram;

public class AMoPercents implements Evaluation<Integer> {
	
	public Integer evaluate(Training training) {
		List<Integer> intervals = training.getIntervals();
		Histogram h = new Histogram(intervals.size()).init();
		for (Integer interval : intervals) {
            h.addRRInterval(interval);
        }
		int maxRangeValue = h.getMaxIntervalNumber();
		int totalCount = h.getTotalCount();
		
		return (int)((maxRangeValue /  (double) totalCount) * 100);
	}

}

File: trunk/PolarMathCS/PolarMath/Evaluation/HRV/AMoPercents.cs
using PolarMath.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PolarMath.Evaluation.Util;

namespace PolarMath.Evaluation.HRV
{
    public class AMoPercents : Evaluation<int>
    {
        public int evaluate(Training training) {
		List<int> intervals = training.getIntervals();
		//Histogram h = new Histogram(intervals.size()).init();
		Histogram h = new Histogram().init();
		foreach (int interval in intervals) {
            h.addRRInterval(interval);
        }
		int maxRangeValue = h.getMaxIntervalNumber();
		int totalCount = h.getTotalCount();
		
		return (int)((maxRangeValue /  (double) totalCount) * 100);
	}
    }
}
